src/components/CreateNewBudgetSheet.js

```diff
@@ -0,0 +1,177 @@
+import { useState } from 'react';
+import firebase from '../firebase';
+import { getDatabase, ref, push } from 'firebase/database';
+import { useNavigate } from 'react-router-dom';
+import WarningModal from './WarningModal';
+
+const NewBudgetSheet = (props) => {
+  const database = getDatabase(firebase);
+  const dbRef = ref(database);
+
+  const [name, setName] = useState('');
+  const [budgetTotal, setBudgetTotal] = useState('');
+  const [categoriesInput, setCategoriesInput] = useState('');
+  const [categoriesArray, setCategoriesArray] = useState([]);
+  const [isWarning, setIsWarning] = useState(false);
+
+  // Forms
+  const handleNameInputChange = (e) => {
+    setName(e.target.value);
+  };
+
+  const handleBudgetInputChange = (e) => {
+    setBudgetTotal(e.target.value);
+  };
+
+  let navigate = useNavigate();
+
+  const handleSubmit = (e, budgetAmount, nameOfSheet, categories) => {
+    e.preventDefault();
+    if (categories.length === 0) {
+      setIsWarning(true);
+    } else {
+      const newBudgetSheet = {
+        budgetCap: budgetAmount,
+        name: nameOfSheet,
+        categories: categories,
+      };
+
+      push(dbRef, newBudgetSheet).then((res) => {
+        setName('');
+        setBudgetTotal('');
+        navigate(`../budget-sheets/${res.key}`);
+      });
+    }
+  };
+
+  const closeModal = () => {
+    setIsWarning(false);
+  };
+
+  // categories
+  const handleCategoryInput = (e) => {
+    setCategoriesInput(e.target.value);
+  };
+
+  const handleAddCategoryClick = (e) => {
+    e.preventDefault();
+    if (categoriesInput && categoriesInput !== ' ') {
```

Comment:
I can circumvent this logic by entering a category of "  ", that is, two spaces. Similarly with more spaces. What about doing something like `if (categoriesInput.trim() !== '')`?

Reply:
fixed by not allowing duplicate categories